libs/ddhub-client-gateway-message-broker/src/lib/services/ddhub-files.service.ts

```diff
@@ -76,6 +76,84 @@ export class DdhubFilesService extends DdhubBaseService {
         }
       );
 
+      const _data = result.data as SendMessageResponseFile;
+      if (_data.recipients.failed > 0) {
+        throw new Error(`upload file with file name: ${originalname} failed`);
+      }
+      this.logger.log(`upload file with file name: ${originalname} successful`);
+      return result.data;
+    } catch (e) {
+      this.logger.error(
+        `upload file with file name: ${originalname} failed`,
+        e
+      );
+      throw e;
+    }
+  }
+
+  @Span('ddhub_mb_uploadFile_chunk')
+  public async uploadFileChunk(
+    file,
+    fileSize: number,
+    chunkSize: number,
+    currentChunkIndex: number,
+    checksum: string,
+    originalname: string,
+    fqcns: string[],
+    topicId: string,
+    topicVersion: string,
+    signature: string,
+    clientGatewayMessageId: string,
+    payloadEncryption: boolean,
+    transactionId?: string
+  ): Promise<SendMessageResponseFile> {
+    try {
+      const formData = new FormData();
+
+      formData.append('file', file);
+      formData.append('fileSize', fileSize);
+      formData.append('chunkSize', chunkSize);
+      formData.append('currentChunkIndex', currentChunkIndex);
+      formData.append('fileChecksum', checksum);
+      formData.append('fileName', originalname);
+      formData.append('fqcns', fqcns.join(','));
+      formData.append('signature', signature);
+      formData.append('topicId', topicId);
+      formData.append('topicVersion', topicVersion);
+      formData.append('clientGatewayMessageId', clientGatewayMessageId);
+      formData.append(
+        'payloadEncryption',
+        payloadEncryption ? 'true' : 'false'
+      );
+
+      if (transactionId) {
+        formData.append('transactionId', transactionId);
+      }
+
+      const result = await this.request<null>(
+        () =>
+          this.httpService.post('/messages/uploadc', formData, {
+            maxContentLength: Infinity,
+            maxBodyLength: Infinity,
+            httpsAgent: this.tlsAgentService.get(),
+            headers: {
+              Authorization: `Bearer ${this.didAuthService.getToken()}`,
+              ...formData.getHeaders(),
+            },
+          }),
+        {
+          stopOnResponseCodes: ['10'],
+        }
+      );
+      if (result.data == "") {
+        this.logger.log(`upload chunk ${currentChunkIndex} file with file name: ${originalname}`);
+        return result.data as SendMessageResponseFile;
```

Comment:
why return empty string?

Reply:
but it's incorrect - you return empty string without any data while telling this is `SendMessageResponseFile` type